htlcswitch/link.go

```diff
@@ -638,7 +638,8 @@ func (l *channelLink) EligibleToForward() bool {
 	return l.channel.RemoteNextRevocation() != nil &&
 		l.ShortChanID() != hop.Source &&
 		l.isReestablished() &&
-		!l.IsDraining(Outgoing)
+		!l.IsDraining(Outgoing) &&
+		l.quiescer.canSendUpdates()
```

Comment:
👍 

Seems like `IsDraining` could just call `canSendUpdates` within the impl? FWIW, missing some partial context from that other PR.

Reply:
There is some slight nuance here where adds are a subset of updates. Intuitively, "draining" means we can't send adds, but we *can* send removes (fulfill/fail) and fee updates. `canSendUpdates` refers to *all* channel updates. If we change `IsDraining` (now `IsFlushing`) to `CanSendAdds` then there are some ways we can consolidate this. I generally try and name things to maximize future readers' ability to understand.